From accc83fdb09b2a7521d17b536433e12f12473c75 Mon Sep 17 00:00:00 2001
From: Fang Hui <b31070@freescale.com>
Date: Thu, 27 Mar 2014 10:59:06 +0800
Subject: [PATCH 506/635] ENGR00304148 - Picture color is wrong for 1M
 pixel/VGA/QVGA with ov5642 sensor

use 1024x768 for 1M pixel, since 720p may sometimes take green picture(5%)

Signed-off-by: Fang Hui <b31070@freescale.com>
---
 mx6/libcamera2/Ov5642Csi.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mx6/libcamera2/Ov5642Csi.cpp b/mx6/libcamera2/Ov5642Csi.cpp
index bfa1ee6..cbf76b1 100755
--- a/mx6/libcamera2/Ov5642Csi.cpp
+++ b/mx6/libcamera2/Ov5642Csi.cpp
@@ -97,8 +97,11 @@ status_t Ov5642Csi::initSensorInfo(const CameraInfo& info)
                 //If w/h ratio is not same with senserW/sensorH, framework assume that
 		//first crop little width or little height, then scale.
 		//But 1920x1080, 176x144 not work in this mode.
+		//For 1M pixel, 720p sometimes may take green picture(5%), so not report it,
+		//use 1024x768 for 1M pixel
 		if( !((vid_frmsize.discrete.width == 1920 && vid_frmsize.discrete.height == 1080) ||
-		      (vid_frmsize.discrete.width == 176 && vid_frmsize.discrete.height == 144))	){
+		      (vid_frmsize.discrete.width == 176 && vid_frmsize.discrete.height == 144) ||
+		      (vid_frmsize.discrete.width == 1280 && vid_frmsize.discrete.height == 720)) ){
 	            mPictureResolutions[pictureCnt++] = vid_frmsize.discrete.width;
 	            mPictureResolutions[pictureCnt++] = vid_frmsize.discrete.height;
 		}
-- 
1.8.0

